### PR TITLE
Added Dependent to Support Page for AB#14643

### PR DESCRIPTION
**Pull request by ytqsl**
# Fixes or Implements [AB#14643](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/14643)

[User Story 14643](https://dev.azure.com/qslvic/Health%20Gateway/_workitems/edit/14643): Admin Delegated Access - Support Page

## Description

Implemented Admin Client support query for dependents and their delegates

## Testing

- [x] Unit Tests Updated
- [x] Functional Tests Updated
- [ ] Not Required

## UI Changes

image

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

**Review by rayking29 — APPROVED** [review]

File: Apps/Common/test/unit/Services/SupportServiceTests.cs [hunk 1]
```diff
@@ -187,6 +196,50 @@ public void ShouldGetVerifications()
             Assert.Equal(2, actualResult.ResourcePayload.Count());
         }
 
+        /// <summary>
+        /// Tests SupportService can get a list of delegate owner by the delegate's PHN.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        public async Task ShouldGetResourceDelegateOwners()
```
Comment (MikeLyttle) [inline Apps/Common/test/unit/Services/SupportServiceTests.cs:204]: This test looks like it's using inverted logic for delegate and dependent.

File: Apps/Admin/Client/Pages/SupportPage.razor.cs [hunk 2]
```diff
@@ -100,7 +100,7 @@ private UserQueryType SelectedQueryType
                 return "Search parameter is required";
             }
 
-            if (this.SelectedQueryType == UserQueryType.Phn && !PhnValidator.Validate(StringManipulator.StripWhitespace(parameter)).IsValid)
+            if ((this.SelectedQueryType == UserQueryType.Phn || this.SelectedQueryType == UserQueryType.Dependent) && !PhnValidator.Validate(StringManipulator.StripWhitespace(parameter)).IsValid)
```
Comment (MikeLyttle) [inline Apps/Admin/Client/Pages/SupportPage.razor.cs:103]: Nit: this expression could be simplified by taking advantage of the PhnOrDependentSelected property added above.

File: Apps/Admin/Client/Pages/SupportPage.razor [hunk 3]
```diff
@@ -50,58 +50,65 @@
...
-            }
-        </ChildRowContent>
-        <PagerContent>
-            <MudTablePager />
-        </PagerContent>
-    </MudTable>
+    <MudGrid>
+        <MudItem xs="12" lg="12">
+            <MudText Class="mt-3" Typo="Typo.subtitle1">
+                <strong>@SelectedQueryType Query Results</strong>
```
Comment (MikeLyttle) [inline Apps/Admin/Client/Pages/SupportPage.razor:56]: `@SelectedQueryType` should be swapped out for `@FormatQueryType(SelectedQueryType)` to achieve the proper casing.

**Review by MikeLyttle — COMMENTED** [review]
There's an inconsistency in the unit tests that should be fixed and some minor things on the front-end that could be improved.

File: Apps/Common/test/unit/Services/SupportServiceTests.cs (hunk 1 above)
Comment (ytqsl) [reply Apps/Common/test/unit/Services/SupportServiceTests.cs:204]: fixed, please re-review

File: Apps/Admin/Client/Pages/SupportPage.razor.cs (hunk 2 above)
Comment (ytqsl) [reply Apps/Admin/Client/Pages/SupportPage.razor.cs]: fixed

File: Apps/Admin/Client/Pages/SupportPage.razor (hunk 3 above)
Comment (ytqsl) [reply Apps/Admin/Client/Pages/SupportPage.razor]: fixed

File: Apps/Common/test/unit/Services/SupportServiceTests.cs [hunk 4]
```diff
@@ -187,6 +196,58 @@ public void ShouldGetVerifications()
             Assert.Equal(2, actualResult.ResourcePayload.Count());
         }
 
+        /// <summary>
+        /// Tests SupportService can get a list of delegate owner by the delegate's PHN.
```
Comment (MikeLyttle) [inline Apps/Common/test/unit/Services/SupportServiceTests.cs:200]: "Tests SupportService can get a list of delegate owner by the delegate's PHN."
=>
"Tests SupportService can get a list of delegates by the dependent's PHN."

File: Apps/Common/test/unit/Services/SupportServiceTests.cs [hunk 5]
```diff
@@ -187,6 +196,58 @@ public void ShouldGetVerifications()
...
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        public async Task ShouldGetResourceDelegateOwners()
```
Comment (MikeLyttle) [inline Apps/Common/test/unit/Services/SupportServiceTests.cs:204]: "ShouldGetResourceDelegateOwners"
=>
"ShouldGetDelegates"

File: Apps/Common/test/unit/Services/SupportServiceTests.cs [hunk 6]
```diff
@@ -187,6 +196,58 @@ public void ShouldGetVerifications()
...
+                        };
+                    });
+
+            ISupportService supportService = CreateSupportService(patientServiceMock: patientServiceMock, resourceDelegateDelegateMock: resourceDelegateDelegateMock);
+
+            RequestResult<IEnumerable<SupportUser>> result = await supportService.GetUsers(UserQueryType.Dependent, dependentPhn).ConfigureAwait(true);
+
+            Assert.Equal(ResultType.Success, result.ResultStatus);
+            Assert.NotNull(result.ResourcePayload);
+            SupportUser[] actualOwners = result.ResourcePayload.ToArray();
```
Comment (MikeLyttle) [inline Apps/Common/test/unit/Services/SupportServiceTests.cs:243]: "actualOwners" => "actualDelegates"

File: Apps/Common/test/unit/Services/SupportServiceTests.cs [hunk 7]
```diff
@@ -187,6 +196,58 @@ public void ShouldGetVerifications()
...
+            Assert.NotEmpty(actualOwners);
+            foreach (string expectedOwnerHdid in expectedDelegateHdids)
```
Comment (MikeLyttle) [inline Apps/Common/test/unit/Services/SupportServiceTests.cs:245]: "expectedOwnerHdid" => "expectedDelegateHdid"

File: Apps/Common/test/unit/Services/SupportServiceTests.cs [hunk 8]
```diff
@@ -187,6 +196,58 @@ public void ShouldGetVerifications()
...
+            {
+                Assert.Contains(actualOwners, owner => owner.Hdid == expectedOwnerHdid);
```
Comment (MikeLyttle) [inline Apps/Common/test/unit/Services/SupportServiceTests.cs:247]: "owner" => "delegate"

File: Apps/Database/src/Delegates/IResourceDelegateDelegate.cs [hunk 9]
```diff
@@ -67,5 +68,33 @@ public interface IResourceDelegateDelegate
...
+        /// </summary>
+        /// <param name="query">the query criteria.</param>
+        /// <returns>A <see cref="Task{TResult}"/> representing the result of the asynchronous operation.</returns>
+        public Task<ResourceDelegateQueryResult> Search(ResourceDelegateQuery query);
+    }
+
+    public record ResourceDelegateQuery
+    {
+        /// <summary>
+        /// Gets or sets search by resource delegate owner's hdid.
```
Comment (MikeLyttle) [inline Apps/Database/src/Delegates/IResourceDelegateDelegate.cs:83]: "Gets or sets search by resource delegate owner's hdid."
=>
"Gets or sets search by resource owner's hdid."

**Review by MikeLyttle — COMMENTED** [review]
The logic for the unit test looks good, but there are still some mixed up references in the documentation and variable names.

**Review by MikeLyttle — APPROVED** [review]

**Review by BrianMaki — APPROVED** [review]

**Pull request merged**